x-pack/plugins/security_solution/public/explore/users/store/middleware_storage.ts

```diff
@@ -5,34 +5,34 @@
  * 2.0.
  */
 
-import type { Action } from 'redux';
-import { map, filter, ignoreElements, tap, withLatestFrom, delay } from 'rxjs/operators';
-import type { Epic } from 'redux-observable';
+import type { Action, Middleware } from 'redux';
 import { get } from 'lodash/fp';
-import type { RootEpicDependencies } from '../../../common/store/epic';
+import type { Storage } from '@kbn/kibana-utils-plugin/public';
+
 import { usersActions } from '.';
 import { selectUserAssetTables } from './selectors';
 import type { UserAssetTableType } from './model';
 import type { State } from '../../../common/store/types';
-export const isNotNull = <T>(value: T | null): value is T => value !== null;
 import { persistUserAssetTableInStorage } from './storage';
 
 const { removeUserAssetTableField, addUserAssetTableField } = usersActions;
 const tableActionTypes = new Set([removeUserAssetTableField.type, addUserAssetTableField.type]);
 
-export const createUserAssetTableLocalStorageEpic =
-  <StateT extends State>(): Epic<Action, Action, StateT, RootEpicDependencies> =>
-  (action$, state$, { storage }) => {
-    const table$ = state$.pipe(map(selectUserAssetTables), filter(isNotNull));
-    return action$.pipe(
-      delay(500),
```

Comment:
I thought this would act similarly to `debounce` but it's actually different. This delay will emit ALL actions, not just the interesting ones, with a gap of each 500ms delay. This delay is not necessary anymore.